server/src/main/java/org/elasticsearch/index/mapper/DataStreamTimestampFieldMapper.java

```diff
@@ -217,8 +218,9 @@ public void postParse(DocumentParserContext context) throws IOException {
         if (first == null) {
             throw new IllegalArgumentException("data stream timestamp field [" + DEFAULT_PATH + "] is missing");
         }
-        TimestampBounds bounds = context.indexSettings().getTimestampBounds();
-        if (bounds != null) {
+        var indexMode = context.indexSettings().getMode();
+        if (indexMode == IndexMode.TIME_SERIES) {
+            TimestampBounds bounds = context.indexSettings().getTimestampBounds();
```

Comment:
This avoids doing a volatile read in case of non tsdb indices. Seems like a good idea?
I think this is the only place where `getTimestampBounds()` getter is being used.

Reply:
I pushed this: 8ae391a1f792adf7f74566ffdd7ba13e52b3a2eb

Reply:
:+1: